hudi-client/src/main/java/org/apache/hudi/index/bloom/HoodieGlobalBloomIndexV2.java

```diff
@@ -0,0 +1,223 @@
+    private IndexFileFilter indexFileFilter;
+    private ExternalSpillableMap<String, BloomFilter> fileIDToBloomFilter;
+    private HoodieTimer hoodieTimer;
+    private long totalTimeMs;
+    private long totalCount;
+    private long totalMetadataReadTimeMs;
+    private long totalRangeCheckTimeMs;
+    private long totalBloomCheckTimeMs;
+    private long totalMatches;
+
+    public LazyRangeAndBloomChecker(Iterator<HoodieRecord<T>> in, final HoodieTable<T> table) {
+      super(in);
+      this.table = table;
+    }
+
+    @Override
+    protected List<Pair<HoodieRecord<T>, String>> computeNext() {
+
+      List<Pair<HoodieRecord<T>, String>> candidates = new ArrayList<>();
+      if (!inputItr.hasNext()) {
+        return candidates;
+      }
+
+      HoodieRecord<T> record = inputItr.next();
+
+      // <Partition path, file name>
+      hoodieTimer.startTimer();
+      Set<Pair<String, String>> matchingFiles = indexFileFilter
+              .getMatchingFilesAndPartition(record.getPartitionPath(), record.getRecordKey());
+
+      totalRangeCheckTimeMs += hoodieTimer.endTimer();
+      hoodieTimer.startTimer();
+
+      matchingFiles.forEach(partitionFileIdPair -> {
+        BloomFilter filter = fileIDToBloomFilter.get(partitionFileIdPair.getRight());
+        if (filter.mightContain(record.getRecordKey())) {
+          totalMatches++;
+          candidates.add(Pair.of(updatePartition(record, partitionFileIdPair.getLeft()), partitionFileIdPair.getRight()));
+        }
+      });
+      totalBloomCheckTimeMs += hoodieTimer.endTimer();
+
+      if (candidates.isEmpty()) {
+        candidates.add(Pair.of(record, ""));
+      }
+
+      totalCount++;
+      return candidates;
+    }
+
+    @Override
+    protected void start() {
+      totalTimeMs = 0;
+      totalMatches = 0;
+      totalCount = 0;
+      hoodieTimer = new HoodieTimer().startTimer();
+
+      // init global range bloomFilter
+      populateFileIDs();
+      populateRangeAndBloomFilters();
+      totalMetadataReadTimeMs += hoodieTimer.endTimer();
+      hoodieTimer.startTimer();
+    }
+
+    @Override
+    protected void end() {
+      totalTimeMs = hoodieTimer.endTimer();
+      String rangeCheckInfo = "LazyRangeAndBloomChecker: "
+              + "totalCount: " + totalCount + ", "
+              + "totalMatches: " + totalMatches + ", "
+              + "totalTimeMs: " + totalTimeMs + "ms, "
+              + "totalMetadataReadTimeMs: " + totalMetadataReadTimeMs + "ms, "
+              + "totalRangeCheckTimeMs: " + totalRangeCheckTimeMs + "ms, "
+              + "totalBloomCheckTimeMs: " + totalBloomCheckTimeMs + "ms";
+      LOG.info(rangeCheckInfo);
+    }
+
+    private HoodieRecord<T> updatePartition(HoodieRecord<T> record, String partitionPath) {
+      HoodieKey hoodieKey = new HoodieKey(record.getKey().getRecordKey(), partitionPath);
+      return new HoodieRecord<>(hoodieKey, record.getData());
+    }
+
+    private void populateFileIDs() {
```

Comment:
is there any difference between regular bloom IndexV2 and global version for this method?